Enabled comparison ops for bfloat16 dtype on CPU

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24224 [WIP] Enabled torch.mm and torch.mv for bfloat16
* #24183 Enabled masked methods for bfloat16
* **#24182 Enabled comparison ops for bfloat16 dtype on CPU**

-----
Fix: Enabled comparison operations for BFloat16 on CPU 
Test: via unit tests

Differential Revision: [D16763460](https://our.internmc.facebook.com/intern/diff/D16763460)

izdeby committed:
Enabled comparison ops for bfloat16 dtype on CPU

gchanan commented on test/test_torch.py:
```diff
@@ -6915,52 +6915,65 @@ def reference(x, k, o3, o32):
+                b = torch.tensor([2], device=device, dtype=dt)
+
+                if dt == torch.bool or dt == torch.half:
+                    continue
```
same as the the other review in the stack, stop doing this.

izdeby committed:
Update on "Enabled comparison ops for bfloat16 dtype on CPU"


-----
Fix: Enabled comparison operations for BFloat16 on CPU 
Test: via unit tests

Differential Revision: [D16763460](https://our.internmc.facebook.com/intern/diff/D16763460)

izdeby committed:
Update on "Enabled comparison ops for bfloat16 dtype on CPU"


-----
Fix: Enabled comparison operations for BFloat16 on CPU 
Test: via unit tests

Differential Revision: [D16763460](https://our.internmc.facebook.com/intern/diff/D16763460)
